library/data/component/src/main/java/org/quiltmc/qsl/component/impl/ComponentsImpl.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.component.impl;
+
+import com.mojang.serialization.Lifecycle;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.Util;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.RegistryKey;
+import net.minecraft.util.registry.SimpleRegistry;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.quiltmc.qsl.component.api.Component;
+import org.quiltmc.qsl.component.api.ComponentInjectionPredicate;
+import org.quiltmc.qsl.component.api.ComponentProvider;
+import org.quiltmc.qsl.component.api.ComponentType;
+import org.quiltmc.qsl.component.impl.sync.ComponentHeaderRegistry;
+import org.quiltmc.qsl.component.impl.util.ErrorUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@ApiStatus.Internal
+public class ComponentsImpl {
+	private static final Logger LOGGER = LoggerFactory.getLogger("QSL/Components");
+	public static final String MODID = "quilt_component";
+    private static final Map<ComponentInjectionPredicate, Set<Identifier>> INJECTION_REGISTRY = new HashMap<>();
+	private static final RegistryKey<Registry<ComponentType<?>>> REGISTRY_KEY =
+			RegistryKey.ofRegistry(new Identifier("quilt", "components"));
+	public static final Registry<ComponentType<?>> REGISTRY =
+			new SimpleRegistry<>(REGISTRY_KEY, Lifecycle.experimental(), null);
```

Comment:
Give these declarations some breathing space!
```suggestion
	private static final Logger LOGGER = LoggerFactory.getLogger("QSL/Components");
	
	public static final String MODID = "quilt_component";
	
    private static final Map<ComponentInjectionPredicate, Set<Identifier>> INJECTION_REGISTRY = new HashMap<>();
    
	private static final RegistryKey<Registry<ComponentType<?>>> REGISTRY_KEY =
			RegistryKey.ofRegistry(new Identifier("quilt", "components"));
	public static final Registry<ComponentType<?>> REGISTRY =
			new SimpleRegistry<>(REGISTRY_KEY, Lifecycle.experimental(), null);
```

Reply:
Okay for this I blame the new intellij version. The new usage fields make it seems like there is space where there actually isn't.

Reply:
...I legitimately forgot those existed, I disabled those like day one :P (right-click on them)